components/telemetry-operator/controller/tracepipeline/controller_test.go

```diff
@@ -36,7 +36,7 @@ var _ = Describe("Deploying a TracePipeline", func() {
 			},
 			Spec: telemetryv1alpha1.TracePipelineSpec{
 				Output: telemetryv1alpha1.TracePipelineOutput{
-					Otlp: telemetryv1alpha1.OtlpOutput{
+					Otlp: &telemetryv1alpha1.OtlpOutput{
```

Comment:
Would probably make more sense to start testing here more than just the minimal setting and asserting that CM and Secret are nil and instead create as much as possible. What do you think?

Reply:
The concurrent test execution of the Ginkgo framework and our limitation to one TracePipeline make it hard to do any comprehensive tests here. So I would prefer to keep testing the content of the generated ConfigMaps, etc. in resources_test.go and limit this test on validating the creation of the Kubernetes objects. But indeed we could do more. I suggest to check for the correct metadata and owner references here.